src/coreclr/vm/interpexec.cpp

```diff
@@ -1186,15 +1186,46 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                     callArgsOffset = ip[2];
                     methodSlot = ip[3];
 
-                    OBJECTREF objRef = AllocateObject((MethodTable*)pMethod->pDataItems[ip[4]]);
+                    MethodTable *pClass = (MethodTable*)pMethod->pDataItems[ip[4]];
```

Comment:
Sounds like this implementation should be a separate opcode since it is polluting the original newobj too much. Also, why isn't this opcode just a simple call reuse like on mono ? `MINT_NEWOBJ_STRING`

Reply:
I'm open to doing it that way. It'll be easier to do with the changes I've been asked to make.